doc/developer/platform/ux.md

```diff
@@ -1,20 +1,945 @@
+schemas in the database.
+
+TODO: should databases be associated with a single [timeline](#timeline)?
+
+### Schema
+
+A schema is a simple container of database objects that can be subject to access
+control. It confers no additional semantics.
+
+The only supported privilege is `CREATE`, which allows users to create objects
+in the schema.
+
+Databases and schemas are simple namespaces to allow for downstream
+organizations and tools to develop opinions about how to structure a Materialize
+Platform project.
+
+For example, an organization might achieve separate production and staging
+environments by creating a `production` database and a `staging` database, then
+using [roles](#role) to limit access to `production` to a trusted set of
+engineers and service accounts.
+
+### Source
+
+The user experience of a source is largely unchanged.
+
+The existing `CREATE MATERIALIZED SOURCE` statement is extended to support
+specifying the cluster in which to create the index:
+
+```
+CREATE MATERIALIZED SOURCE <name> FROM ... [IN CLUSTER <cluster>]
+```
+
+If no cluster is explicitly specified, the active cluster is used. If there is
+no active cluster, the statement is rejected.
+
+Users cannot read data from a source on which they do not have the `SELECT`
+privilege.
+
+Note that if the source references a [secret](#secret), privileges on that
+secret are *not* considered when determining whether a user can read the data in
+the source. In other words, the `SELECT` privilege on the source grants access
+to the data protected by the secret, but not to the secret itself.
+
+Users cannot create a source if they do not have the `CREATE SOURCE` permission
+in the region.
+
+### Sink
+
+Like indexes, sinks cause computation to occur, and therefore must be bound to a
+cluster.
+
+Users cannot create a sink if they do not have the `CREATE SOURCE` permission
+in the region.
+
+We do not yet know how to provide sinks in Materialize Platform. We are,
+however, aware of at least two important user experience constraints:
+
+  * The default sink experience must be "exactly once." If a cluster restarts,
+    downstream sinks must not reexport data that was previously exported.
+
+  * Sinks must be prepared for the behavior of the COMPUTE layer to change after
+    a version upgrade. If we fix a bug in COMPUTE that changes the data computed
+    by an index, sinks must detect and communicate this behavior change to
+    users. Options include marking the sink as errored and asking users to
+    recreate the sink, or emitting the diff between the old computation and
+    new computation and then proceeding with export of new values.
+
+### Table
+
+Unlike tables in `materialized`, tables in Materialize Platform will durably
+store their data. They will no longer have a mandatory primary index; users
+can leave their tables as unindexed storage collections, if they so choose.
+
+Users cannot read data from a table on which they do not have the `SELECT`
+privilege.
```

Comment:
Let's specify what the context for a table is. Is it tied to a region?

Reply:
That's right! The context of each of these objects (including tables) is described up in the overview section.